### DISCUSSION TITLE
__console_patch prop in the new Console

### DISCUSSION THREAD

**Discussion by mirjammalksoo** [twilio-labs/paste#1356] [Q&A] [answered]
Hi!

We are using  `__console_patch` prop for `Toast` and `Modal` in our Console pages but we are experiencing a different behaviour inside old (twilio.com) and new Console (console.twilio.com, in iframe). For example, this results in having our pages shifted down when opening modals in an iframe because as I understand, the prop is only meant to fix the issues in the old console. 

It seems this will affect all Console pages using this prop that are living in an iframe and have not yet migrated to the new Console repository.

I am wondering if you are aware of this behaviour and will there be some kind of solution for this? 

From the [`__console_patch` PR](https://github.com/twilio-labs/paste/pull/1287):
`It is only intended to be used in the Console product, which is why the prop name is described with a two underscore prefix. It should not be used outside of the Console product and doesn't affect our other customers.`

Does Console product mean the old Console or also the new Console?

Thanks!

**Comment by TheSisb** [comment, marked as answer]
Hi @mirjammalksoo,

This prop was created before the new console was created, so I can see how the documentation is now confusing. We will update the documentation to more clearly mention the old console product. Please do not use this prop in the new console.

Thanks for bringing this to our attention!

**Reply by mirjammalksoo** [reply]
Hi @TheSisb!

Thank you for the explanation!

However currently our pages/code exist in both consoles - in the old console and in the new console's iframe. This applies to most of the projects which are using Paste but haven't migrated yet to the new Console repository.

To clarify our next steps for resolving this issue, is this something that every project is expected to handle separately, i.e. dynamically decide based on the Console type (old vs new) whether to add the `__console_patch` property or not?

Thanks!

**Reply by TheSisb** [reply]
Thanks for the clarification @mirjammalksoo and good point!  I started a [discussion on Slack](https://twilio.slack.com/archives/C01DLUML16K/p1618326633139000) with the console team if you wanted to follow along.
